### PR TITLE
Curated Package Documentation

### PR DESCRIPTION
**Pull request by vincentni**
*Issue #, if available:*

*Description of changes:*

*Testing (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/content/en/docs/tasks/packages/_index.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+---
+title: "Package management"
+date: 2022-04-12
+weight: 20
+description: >
+  Common tasks for managing curated packages.
+---
+
+The main goal of EKS Anywhere curated packages is to make it easy to install, configure and maintain operational components in an EKS Anywhere cluster. EKS Anywhere curated packages offers to run secure and tested operational components on EKS Anywhere clusters. Please check out [package controller]({{< relref "../../concepts" >}}) for more details.
```
Comment (vincentni) [inline docs/content/en/docs/tasks/packages/_index.md:9]: This needs to be updated to link to the page under https://github.com/aws/eks-anywhere/pull/1877.

File: docs/content/en/docs/getting-started/local-environment/_index.md [hunk 2]
```diff
@@ -71,11 +71,23 @@ The configuration specification is divided into two sections:
 
 For full EKS Anywhere configuration reference for a VMware vSphere cluster and explanation on each parameter in the configuration generated above refer vSphere configuration
 
+1. Generate a curated-packages config
+   {{% alert title="Note" color="primary" %}}
+   * It is *optional* to install the curated packages as part of the cluster creation and `eksctl anywhere` version should be `v0.9.0` or later.
```
Comment (TerryHowe) [inline docs/content/en/docs/getting-started/local-environment/_index.md:76]:
```suggestion
   * It is *optional* to install the curated packages as part of the cluster creation
   * `eksctl anywhere version` version should be `v0.9.0` or later.
```

File: docs/content/en/docs/getting-started/local-environment/_index.md [hunk 3]
```diff
@@ -71,11 +71,23 @@ The configuration specification is divided into two sections:
...
+   * Post-creation installation and detailed package configurations can be found [here.]({{< relref "../../tasks/packages" >}})
+   {{% /alert %}}
+   Example shows how to install two packages `flux` and `harbor` from the [curated package list]({{< relref "../../reference/packagespec" >}}).
```
Comment (TerryHowe) [inline docs/content/en/docs/getting-started/local-environment/_index.md:79]:
```suggestion
   Example shows how to install two packages `harbor` from the [curated package list]({{< relref "../../reference/packagespec" >}}).
```

File: docs/content/en/docs/getting-started/local-environment/_index.md [hunk 4]
```diff
@@ -71,11 +71,23 @@ The configuration specification is divided into two sections:
...
+   ```bash
+   eksctl anywhere generate package flux harbor -d .
```
Comment (TerryHowe) [inline docs/content/en/docs/getting-started/local-environment/_index.md:81]:
```suggestion
   eksctl anywhere generate package harbor -d .
```

File: docs/content/en/docs/getting-started/local-environment/_index.md [hunk 5]
```diff
@@ -71,11 +71,23 @@ The configuration specification is divided into two sections:
...
+   ```
 
 1. Create a cluster
 
    ```bash
+   # Create a cluster without curated packages installation
    eksctl anywhere create cluster -f $CLUSTER_NAME.yaml
+   # Create a cluster with curated packages installation
+   eksctl anywhere create cluster -f $CLUSTER_NAME.yaml --install-packages ./curated-packages/
```
Comment (TerryHowe) [inline docs/content/en/docs/getting-started/local-environment/_index.md:90]: Can't we use a default directory here. I think the packages directory would have to be pretty much hard coded for gitops

File: docs/content/en/docs/getting-started/local-environment/_index.md [hunk 6]
```diff
@@ -128,4 +140,8 @@ For full EKS Anywhere configuration reference for a VMware vSphere cluster and e
    ```
 
    Verify the test application in the [deploy test application section]({{< relref "../../tasks/workload/test-app" >}}).
-   See the [Cluster management]({{< relref "../../tasks/cluster" >}}) section with more information on common operational tasks like scaling and deleting the cluster.
+
+## Next steps:
+* See the [Cluster management]({{< relref "../../tasks/cluster" >}}) section with more information on common operational tasks like scaling and deleting the cluster.
+
+* See the [Package management]({{< relref "../../tasks/packages" >}}) section with more information on post-creation curated packages installation.
```
Comment (TerryHowe) [inline docs/content/en/docs/getting-started/local-environment/_index.md:147]:
```suggestion
* See the [Package management]({{< relref "../../tasks/packages" >}}) section for more information on post-creation curated packages installation.
```

File: docs/content/en/docs/getting-started/local-environment/_index.md (hunk 3 above)
Comment (TerryHowe) [reply docs/content/en/docs/getting-started/local-environment/_index.md:79]: We aren't supporting flux as far as I know. Same comment applies to prod section below

File: docs/content/en/docs/tasks/packages/harbor/_index.md [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+To install package controller, please follow the [installation guide.]({{< relref ".." >}})
+
+{{% /alert %}}
+
+## Install
+
+<!-- this content needs to be indented so the numbers are automatically incremented -->
+1. Generate the package configuration
+   ```bash
+   eksctl anywhere generate package Harbor --source cluster -d .
```
Comment (TerryHowe) [inline docs/content/en/docs/tasks/packages/harbor/_index.md:21]:
```suggestion
   eksctl anywhere generate package harbor --source cluster -d .
```
This should be all case insensitive

**Review by TerryHowe — CHANGES_REQUESTED** [review]
I have more, this is a start

File: docs/content/en/docs/workshops/introduction/harbor.md [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+
+   * `harbor.eksa.demo:30003` should be replaced with whatever `externalURL` is set to in the Harbor package yaml file.
+
+   {{% /alert %}}
+    ```bash
+    docker pull harbor.eksa.demo:30003/proxy-private-project/alpine:latest
+    ```
+
+## Repository replication from Harbor to a private ECR repository
+This use case is to use Harbor to replicate local images and charts to a private ECR repository in a push mode. When a replication rule is set, all resources that match the defined filter patterns are replicated to the destination registry when the triggering condition is met.
```
Comment (ewollesen) [inline docs/content/en/docs/workshops/introduction/harbor.md:122]:
```suggestion
This use case is to use Harbor to replicate local images and charts to a private ECR repository in push mode. When a replication rule is set, all resources that match the defined filter patterns are replicated to the destination registry when the triggering condition is met.
```

File: docs/content/en/docs/workshops/introduction/harbor.md [hunk 9]
```diff
@@ -0,0 +1,214 @@
...
+    ```
+
+1. Authenticate with Harbor with the default credential as shown below
+    ```bash
+    admin
+    Harbor12345
+    ```
+   {{% alert title="Note" color="primary" %}}
+
+   * `harbor.eksa.demo:30003` should be replaced with whatever `externalURL` is set to in the Harbor package yaml file.
```
Comment (ewollesen) [inline docs/content/en/docs/workshops/introduction/harbor.md:188]:
```suggestion
   * `harbor.eksa.demo:30003` should be replaced with whatever `externalURL` is set to in the Harbor package YAML file.
```
This is a little detail, but YAML is an acronym, so anywhere in the document that you're not using YAML as part of a filename, it should be capitalized. I mean, people are gonna know what you mean either way, but it looks more professional/better capitalized.

**Review by ewollesen — COMMENTED** [review]
WOW! Great job Vincent! This was a _TON_ of good work!

I can't leave a comment directly, but in `docs/static/images/harbor-private_proxy_project.png`, `docs/static/images/harbor-replication_result.png`, and `docs/static/images/harbor-replication_rule.png` an account ID is visible. It should probably be blurred out.

**Comment by TerryHowe** [comment]
BTW, you can just commit the suggestions and git pull your branch locally to get all the updates. That would make our reviewing easier for this large PR

File: docs/content/en/docs/tasks/packages/harbor/_index.md [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+linkTitle: "Add Harbor"
+weight: 13
+date: 2022-04-12
+description: >
+  Install/upgrade/uninstall Harbor
+---
+
+{{% alert title="Important" color="warning" %}}
+
+To install package controller, please follow the [installation guide.]({{< relref ".." >}})
```
Comment (TerryHowe) [inline docs/content/en/docs/tasks/packages/harbor/_index.md:12]: I don't think this needs to be an alert 
```suggestion
If your cluster was created with a release of EKS Anywhere prior to 0.9.0, you may need to [installation the package controller.]({{< relref ".." >}})
```

File: docs/content/en/docs/tasks/packages/_index.md [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+
+{{% alert title="Important" color="warning" %}}
+
+* To install EKS Anywhere, create an EKS Anywhere cluster, or review the EKS Anywhere system requirements, see the [getting-started guide.]({{< relref "../../getting-started" >}})
```
Comment (chrisnegus) [inline docs/content/en/docs/tasks/packages/_index.md:13]:
```suggestion
* To install EKS Anywhere, create an EKS Anywhere cluster or review the EKS Anywhere system requirements.
See the [Getting started]({{< relref "../../getting-started" >}}) guide for details.
```

File: docs/content/en/docs/tasks/packages/harbor/_index.md [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+   Please see [complete configuration options]({{< relref "../../../reference/packagespec/harbor" >}}) for all configuration options and their default values.
+
+   {{% alert title="Important" color="warning" %}}
+   * All configuration options are listed in dot notations (e.g., `expose.tls.enabled`) in the doc, but they have to be transformed to **hierachical structures** when specified in the `config` section in the YAML spec.
+   * Harbor web portal is exposed through `NodePort` by default, and its default port number is `30003` with TLS enabled and `30002` with TLS disabled.
+   * TLS is enabled by default for connections to Harbor web portal, and a secret resource named `harbor-tls-secret` is required for that purpose. It can be provisioned through cert-manager or manually with the following command using self-signed certificate:
+      ```bash
+      kubectl create secret tls harbor-tls-secret --cert=[path to certificate file] --key=[path to key file] -n eksa-packages
+      ```
+   * `secretKey` has to be set as a string of 16 chars for encryption.
```
Comment (chrisnegus) [inline docs/content/en/docs/tasks/packages/harbor/_index.md:35]:
```suggestion
   * `secretKey` has to be set as a string of 16 characters for encryption.
```

File: docs/content/en/docs/workshops/introduction/harbor.md [hunk 13]
```diff
@@ -0,0 +1,214 @@
...
+    ```bash
+    admin
+    Harbor12345
+    ```
+ 
+    ![Harbor web portal](/images/harbor-portal.png)
+
+1. Create a registry proxy
+
+    Navigate to `Registries` on the left panel, and then click on `NEW ENDPOINT` button. Choose `Docker Registry` as Provider, and enter `public-ecr` as Name, `https://public.ecr.aws/` as Endpoint URL. Save it by click on OK.
```
Comment (chrisnegus) [inline docs/content/en/docs/workshops/introduction/harbor.md:30]:
```suggestion
    Navigate to `Registries` on the left panel, and then click on `NEW ENDPOINT` button. Choose `Docker Registry` as the Provider, enter `public-ecr` as the Name, and enter `https://public.ecr.aws/` as the Endpoint URL. Save it by clicking on OK.
```

File: docs/content/en/docs/workshops/introduction/harbor.md [hunk 14]
```diff
@@ -0,0 +1,214 @@
...
+
+    ![Harbor public ecr proxy](/images/harbor-public_ecr_proxy.png)
+
+1. Create a proxy project
+
+    Navigate to `Projects` on the left panel and click on `NEW PROJECT` button. Enter `proxy-project` as Project Name, check `Public access level`, and turn on Proxy Cache and choose `public-ecr` from the pull-down list. Save the configuration by clicking on OK.
```
Comment (chrisnegus) [inline docs/content/en/docs/workshops/introduction/harbor.md:36]:
```suggestion
    Navigate to `Projects` on the left panel and click on the `NEW PROJECT` button. Enter `proxy-project` as the Project Name, check `Public access level`, turn on Proxy Cache and choose `public-ecr` from the pull-down list. Save the configuration by clicking on OK.
```

File: docs/content/en/docs/workshops/introduction/harbor.md [hunk 15]
```diff
@@ -0,0 +1,214 @@
...
+ 
+    ![Harbor web portal](/images/harbor-portal.png)
+
+1. Create a registry proxy
+
+    In order for Harbor to proxy a remote private ECR registry, an IAM credential with necessary permissions need to be created. Usually, it follows three steps: 
+
+    1. Policy
+
+        This is where you specify all necessary permissions and please refer to [private repository policies](https://docs.aws.amazon.com/AmazonECR/latest/userguide/repository-policies.html), [IAM permissions for pushing an image](https://docs.aws.amazon.com/AmazonECR/latest/userguide/image-push.html) and [ECR policy examples](https://docs.aws.amazon.com/AmazonECR/latest/userguide/security_iam_id-based-policy-examples.html) to figure out the minimal set of required permissions.
```
Comment (chrisnegus) [inline docs/content/en/docs/workshops/introduction/harbor.md:69]:
```suggestion
        This is where you specify all necessary permissions. Please refer to [private repository policies](https://docs.aws.amazon.com/AmazonECR/latest/userguide/repository-policies.html), [IAM permissions for pushing an image](https://docs.aws.amazon.com/AmazonECR/latest/userguide/image-push.html) and [ECR policy examples](https://docs.aws.amazon.com/AmazonECR/latest/userguide/security_iam_id-based-policy-examples.html) to figure out the minimal set of required permissions.
```

File: docs/content/en/docs/workshops/introduction/harbor.md [hunk 16]
```diff
@@ -0,0 +1,214 @@
...
+
+   ![Harbor nonproxy project](/images/harbor-nonproxy_project.png) 
+
+1. Create a registry proxy
+
+    In order for Harbor to proxy a remote private ECR registry, an IAM credential with necessary permissions need to be created. Usually, it follows three steps: 
+
+    1. Policy
+
+        This is where you specify all necessary permissions and please refer to [private repository policies](https://docs.aws.amazon.com/AmazonECR/latest/userguide/repository-policies.html), [IAM permissions for pushing an image](https://docs.aws.amazon.com/AmazonECR/latest/userguide/image-push.html) and [ECR policy examples](https://docs.aws.amazon.com/AmazonECR/latest/userguide/security_iam_id-based-policy-examples.html) to figure out the minimal set of required permissions.
```
Comment (chrisnegus) [inline docs/content/en/docs/workshops/introduction/harbor.md:144]:
```suggestion
        This is where you specify all necessary permissions. Please refer to [private repository policies](https://docs.aws.amazon.com/AmazonECR/latest/userguide/repository-policies.html), [IAM permissions for pushing an image](https://docs.aws.amazon.com/AmazonECR/latest/userguide/image-push.html) and [ECR policy examples](https://docs.aws.amazon.com/AmazonECR/latest/userguide/security_iam_id-based-policy-examples.html) to figure out the minimal set of required permissions.
```

File: docs/content/en/docs/workshops/introduction/harbor.md [hunk 17]
```diff
@@ -0,0 +1,214 @@
...
+
+        ![Harbor private ecr user group](/images/harbor-private_ecr_user_group.png)
+
+    1. User
+
+        Create a user and add it to the user group. In addition, please navigate to Security credentials to generate an access key. Access keys consists of two parts: an access key ID and a secret access key. Please save both as they are used in the next step. 
+
+        ![Harbor private ecr user](/images/harbor-private_ecr_user.png)
+
+    Navigate to `Registries` on the left panel, and then click on `NEW ENDPOINT` button. Choose `Aws ECR` as Provider, and enter `private-ecr` as Name, `https://[ACCOUNT NUMBER].dkr.ecr.us-west-2.amazonaws.com/` as Endpoint URL, use the access key ID part of the generated access key as Access ID, and use the secret access key part of the generated access key as Access Secret. Save it by click on OK.
```
Comment (chrisnegus) [inline docs/content/en/docs/workshops/introduction/harbor.md:162]:
```suggestion
    Navigate to `Registries` on the left panel, and then click on the `NEW ENDPOINT` button. Choose `Aws ECR` as the Provider, enter `private-ecr` as the Name, and `https://[ACCOUNT NUMBER].dkr.ecr.us-west-2.amazonaws.com/` as the Endpoint URL, use the access key ID part of the generated access key as Access ID, and use the secret access key part of the generated access key as Access Secret. Save it by clicking on OK.
```

**Review by chrisnegus — COMMENTED** [review]
Good content! I have a few suggestions.

File: docs/content/en/docs/tasks/packages/_index.md [hunk 18]
```diff
@@ -0,0 +1,44 @@
...
+
+{{% alert title="Important" color="warning" %}}
+
+* To install the EKS Anywhere binaries, create a EKS Anywhere cluster and see system requirements, please follow the [getting-started guide.]({{< relref "../../getting-started" >}})
+
+* Check the existence of `packagecontroller` with the following commands:
+    ```bash
+    kubectl get packagecontroller -n eksa-packages
+    ```
+* Check if the version of `eksctl anywhere` is above `v0.9.0` with the following commands:
```
[the remark this answers is not in the document]
Comment (TerryHowe) [reply docs/content/en/docs/tasks/packages/_index.md:22]:
```suggestion
* Check if the version of `eksctl anywhere` is `v0.9.0` or above with the following commands:
```

**Review by chrisnegus — APPROVED** [review]
/lgtm

**Pull request merged**